### PR TITLE
Adding ITs to check for Default Role Permissions

### PR DESCRIPTION
**Pull request by aditjind**
Signed-off-by: Aditya Jindal <aditjind@amazon.com>

*Issue #, if available:* https://github.com/opensearch-project/alerting/issues/219

*Description of changes:*

*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by aditjind** [comment]
ITs tests passing locally: - https://github.com/aditjind/alerting/pull/12

**Review by AWSHurneyt — APPROVED** [review]

File: alerting/src/test/kotlin/org/opensearch/alerting/TestHelpers.kt [hunk 1]
```diff
@@ -218,7 +216,7 @@ val NEVER_RUN = Script("return false")
 val DRYRUN_MONITOR = mapOf("dryrun" to "true")
 val TEST_HR_INDEX = "hr_data"
 val TEST_NON_HR_INDEX = "not_hr_data"
-val TEST_HR_ROLE = "hr_role"
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/TestHelpers.kt]: Can we keep this as `hr_role`? Is there a reason why we want to change it?

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt [hunk 2]
```diff
@@ -128,6 +131,79 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
...
+            fail("Expected 403 Method FORBIDDEN response")
+        } catch (e: ResponseException) {
+            assertEquals("Unexpected status", RestStatus.FORBIDDEN, e.response.restStatus())
+        } finally {
+            deleteRoleMapping(TEST_HR_ROLE)
+            deleteRole(TEST_HR_ROLE)
+        }
+    }
+
+    fun `test query monitors with an user with read only role`() {
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt:161]: Update test name to indicate this is a user with search monitor cluster permission only.
For example: `test query monitors with an user with only search monitor cluster permission`

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt [hunk 3]
```diff
@@ -128,6 +131,79 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
...
+        )
+
+        assertEquals("Search monitor failed", RestStatus.OK, searchResponse.restStatus())
+        val xcp = createParser(XContentType.JSON.xContent(), searchResponse.entity.content)
+        val hits = xcp.map()["hits"]!! as Map<String, Map<String, Any>>
+        val numberDocsFound = hits["total"]?.get("value")
+        assertEquals("Monitor not found during search", 1, numberDocsFound)
+    }
+
+    fun `test query monitors with an user without read only role`() {
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt:186]: Change the last part to `without search monitor cluster permission`.

File: alerting/src/test/kotlin/org/opensearch/alerting/TestHelpers.kt (hunk 1 above)
Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/TestHelpers.kt]: done

File: alerting/src/test/kotlin/org/opensearch/alerting/AlertingRestTestCase.kt [hunk 4]
```diff
@@ -828,6 +828,34 @@ abstract class AlertingRestTestCase : ODFERestTestCase() {
...
+                ]
+              }],
+              "tenant_permissions": []
+            }
+        """.trimIndent()
+        request.setJsonEntity(entity)
+        logger.debug("Input Request to the Create Custom Role: $request")
+        logger.error("Input Request to the Create Custom Role: $request")
+        logger.info("Input Request to the Create Custom Role: $request")
+        logger.warn("Input Request to the Create Custom Role: $request")
```
[the remark this answers is not in the document]
Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/AlertingRestTestCase.kt]: Missed removing them.

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt (hunk 2 above)
Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt]: done

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt (hunk 3 above)
Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt]: done

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt [hunk 5]
```diff
@@ -430,6 +506,34 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
         }
     }
 
+    fun `test ack alerts with an user with ack alerts role`() {
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt:509]: This test is only checking if the user with `ALERTING_ACK_ALERTS` role can view the alerts and not acknowledging the alerts. Please check for that.

Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt:509]: Will add as a part of another test in the next PR?

Comment (aditjind) [reply alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt]: Renamed the test as `test get alerts with an user with get alerts role` and changed the role to get alerts role.

**Review by lezzago — APPROVED** [review]

**Pull request merged**